contracts/utils/Nonces.sol

```diff
@@ -27,4 +32,15 @@ abstract contract Nonces {
             return _nonces[owner]++;
         }
     }
+
+    /**
+     * @dev Same as {_useNonce} but checking that `nonce` is the next valid for `owner`.
+     */
+    function _useCheckedNonce(address owner, uint256 nonce) internal virtual returns (uint256) {
+        uint256 current = _useNonce(owner);
+        if (nonce != current) {
+            revert InvalidAccountNonce(owner, current);
+        }
+        return current;
+    }
```

Comment:
This function checks that `nonce` (provided as an argument) is equal to `current`. That means if this function does not revert, the caller known what the value is already. As a consequence, there is not much value to returning the current.

I'd simplify the code by doing 
```suggestion
    function _useCheckedNonce(address owner, uint256 nonce) internal virtual {
        if (nonce != _useNonce(owner)) {
            revert InvalidAccountNonce(owner, current);
        }
    }
```

Reply:
It won't work because `current` is not defined, we'll need to declare `uint256 current` and I don't see why not return it.

I'd rather have it than not, does this change your opinion?

Reply:
My bad, I forgot about the fact that we need to declare current to put it in the error :/